statick_tool/statick.py

```diff
@@ -579,30 +580,45 @@ def scan_package(
         num_packages: int,
     ) -> Optional[Dict[str, List[Issue]]]:
         """Scan each package in a separate process while buffering output."""
+        logger = logging.getLogger()
+        old_handler = None
+        if logger.handlers[0]:
+            old_handler = logger.handlers[0]
+            handler = MemoryHandler(10000, flushLevel=logging.ERROR, target=old_handler)
```

Comment:
the `10000` is how many log statements it will buffer before automatically flushing to the target handler. I chose a number that I thought would not be easily reached by a single call to `run`.
the `flushLevel` indicates that if a log statement comes it at the `ERROR` level the handler will immediately flush it's buffer. it seemed to me that for how we use the log levels, anything error and above should be immediately printed. if that should be changed we have two options:
1. up that to critical, which we don't currently use.
2. create our own subclass of the BufferingHandler that implements the `shouldFlush` function as we want
https://docs.python.org/3/library/logging.handlers.html#memoryhandler

Reply:
The `10000` number seems fine to me. That is per package, right? I have only occasionally seen that many issues for large workspaces, so I think it will be fine either way.

I like your approach to printing out `ERROR` and higher immediately. I think that is a great place to start. If we or other users find that a different approach should be taken then we can re-visit the other options you listed.

Reply:
Yes. It'll be per call to `scan_package`, which is called with a single package once per subprocess.